bestie/proto/test_metrics.proto

```diff
@@ -5,6 +5,42 @@ package bestie.proto;
 
 option go_package = "github.com/enfabrica/enkit/bestie/proto";
 
+// Each metric tag must be supplied in the form of a key/value string pair.
+message KeyValuePair {
+    string key = 1;
+    string value = 2;
+}
+
+// Specialized BigQuery test result definition compatible with Prometheus remote read adapter.
+//
+// This message is similar to TestMetric, but with added restrictions to make it
+// less ambiguous for the message sender to construct properly. The same BigQuery
+// table holds both TestMetric and ResultMetric rows from a variety of test applications
+// (clients), therefore standardizing on a specific metricname ('testresult') is
+// essential. This name, a corresponding 'result' tag, and a numeric value is inserted
+// by the BES Endpoint service as it processes a ResultMetric message, therefore these are
+// not specifically defined by the message fields.
+//
+// The allowed Result values are PASS or FAIL, with INCOMPLETE indicating a test
```

Comment:
pass and fail for a test should already be captured either by the normal BES protocol, or by parsing the test.xml file in the stream that contains pass/fail information for individual tests?

if we keep the result, can we define a ResultMetric as a TestMetric with on optional Result? eiter as:
```
message ResultMetric {
  ...
  Result result;
  TestMetric metric;
}
```

or even just add the `Result` field in the `TestMetric` directly? set to an "UNSET" default value? so a test setting it explicitly marks it as FAIL/PASS/SKIP/INCOMPLETE, a test that does not know about the field leaves the default 0 value of UNSET?

Reply:
The reason I defined ResultMetric separate from TestMetric is that they convey different information. A single test case can produce multiple TestMetrics, but only one ResultMetric (e.g. pass/fail). Since these end up in the same BigQuery 'testmetrics' table, the ResultMetric needs a distinct metricname identifier "testresult" to allow querying. Instead of relying on the test application to spell it right, the ResultMetric protobuf msg allows the BES Endpoint to take care of the metricname for consistency.

I'm not using the test.xml file, as it does not carry enough useful information for creating BigQuery table entries. Hence, I'm relying on test_metrics.proto and the resultant *.metrics.pb file to give us what we need, which we can control.

Reply:
> Instead of relying on the test application to spell it right

...my vote is to rely on the test application to spell it right. The test application should be going through a test metrics client library which emits this file, so really it's actually the client library's responsibility to emit the testresult metric with the proper spelling.

Doing it the way proposed in this PR necessitates:
* client libraries emitting a special result metric anyway
* all this extra schema surface area to understand/code around

Doing it where client libraries know to emit a specially named metric - well, the libraries have to special-case that metric either way, so might as well avoid all this code